### PR TITLE
User management docs review

### PR DESCRIPTION
**Pull request by tiagojsag**

### CODE REVIEW

File: source/includes/api/_authentication.md [hunk 1]
```diff
@@ -1,46 +1,54 @@
...
+Once you have your token, you can use it to authenticate your requests to the RW API. Depending on which endpoints you are trying to use, certain actions may become available by simply authenticating your request. Other will require your account to have a special role or applications, which we'll cover in more detail in the [User Management](#user-management) documentation.
+
+Tokens must be provided in HTTP request header, named as `Authorization`, with the structure `Bearer: <your token>`.
+
+Once generated, a token is valid until any of the associated user information (name, email, or associated applications) changes. If your token becomes invalid, you will need to log in and go to [https://api.resourcewatch.org/auth/generate-token](https://api.resourcewatch.org/auth/generate-token) to generate a new token.
+
+
+## More information on authentication and user management
+
+This section's goal is to help you learn the very basics of how you can quickly create a user account and use your token to authenticate requests. For more in-depth details on authentication, token and user management, refer to the [User Management](#user-management) section of the documentation.
```
Comment (ikas) [inline source/includes/api/_authentication.md]: IMO this paragraph should be placed at the start of the Authentication section, no need to have an extra section just for this.

File: source/includes/api/_user_management.md [hunk 2]
```diff
@@ -293,98 +452,182 @@ curl -X GET http://api.resourcewatch.org/auth/user
...
 ```
 
+Lists user accounts:
+
+- Only users with role ADMIN have permissions to use this endpoint.
+- Email+password based accounts that have not had their email address confirmed will not be listed by this endpoint.
+- By default, only users belonging to the same apps as the requesting user will be shown. See [Filter by app](#filter-by-app) for more options.
+
 
 #### Filter by name
```
Comment (ikas) [inline source/includes/api/_user_management.md]: Is it worth having a separate section for each possible query param filter? Why not defining all the available query param filters in a table, like we did for other API resources? :)

**Review by ikas — CHANGES_REQUESTED** [review]
Some minor organizational comments, other than that looks great to me :+1:

File: source/includes/api/_authentication.md (hunk 1 above)
Comment (tiagojsag) [reply source/includes/api/_authentication.md]: good point, will change it

File: source/includes/api/_user_management.md (hunk 2 above)
Comment (tiagojsag) [reply source/includes/api/_user_management.md]: apparently I missed this section in my review :/

File: source/includes/api/_user_management.md [hunk 3]
```diff
@@ -1,47 +1,75 @@
...
+
+`application` is a list of keys meant to identify the different client applications that are built using the RW API. It's present not only on user accounts, but also on many of the resources found on the RW API, either as a single value or as a list of values. Typically, in order to manipulate a given resource, that resource and the user account must have at least one overlapping `application` value.
+
+Keep in mind that it's up to each individual RW API service (dataset, widget, layer, etc) to define how they restrict or allow actions based on these or other factors, so the examples above may not be true for all cases. Refer to the documentation of each resource and endpoint for more details on restrictions they may have regarding user accounts and their properties.
```
Comment (ethanroday) [inline source/includes/api/_user_management.md:29]: On a scale of 1-10, how good do we feel about how comprehensive the role restriction documentation is throughout these docs?
Separately, consistent would you say the definitions of user, manager, and admin are?

File: source/includes/api/_user_management.md [hunk 4]
```diff
@@ -57,65 +85,132 @@ curl -X POST http://api.resourcewatch.org/auth/login \
...
+```shell
+curl -X GET "https://api.resourcewatch.org/auth/generate-token" \
+    -H 'Cookie: <your cookie values>'
+```
+
+> Response:
+
+```json
+{
+    "token": "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6IjViZWNmYTJiNjdkYTBkM2VjMDdhMjdmNiIsInJvbGUiOiJVU0VSIiwicHJvdmlkZXIiOiJsb2NhbCIsImVtYWlsIjoidGVzdEBleGFtcGxlLmNvbSIsImV4dHJhVXNlckRhdGEiOnsiYXBwcyI6WyJydyJdfSwiY3JlYXRlZEF0IjoxNTQzMzE1NzMxNzcwLCJpYXQiOjE1NDMzMTU3MzF9.kIdkSOb7mCMOxE2ipqVOBrK7IefAjLDhaPG9DT1qvCw"
```
Comment (ethanroday) [inline source/includes/api/_user_management.md:158]: This isn’t anyone’s actual token, right?

File: source/includes/api/_user_management.md [hunk 5]
```diff
@@ -155,35 +249,40 @@ curl -X POST http://api.resourcewatch.org/auth/sign-up?origin=rw \
...
-It accepts an optional `callbackUrl` query parameter with an URL to which the user will be redirect if the confirmation succeeds.
+While optional, it's highly recommended that you specify which apps the user will be granted access to, as most API operation validate the user's apps match datasets, widgets, etc. All accounts created this way will have the `USER` role.
 
-Should no `callbackUrl` be provided, the user is redirected to an URL based on the first application associated to their user account - see `ct-oauth-plugin` configuration for more info.
+**Errors**
+
+Error code     | Error message  | Description
+-------------- | -------------- | --------------
+422            | Email exists.  | An account already exists for the provided email address.
+422            | Email, Password and Repeat password are required.  | You are missing one of the required fields.
```
Comment (ethanroday) [inline source/includes/api/_user_management.md:271]: Same code as the previous one?

File: source/includes/api/_user_management.md [hunk 6]
```diff
@@ -293,98 +454,179 @@ curl -X GET http://api.resourcewatch.org/auth/user
 }
 ```
 
+This endpoint allows users with `ADMIN` role to list and filter existing user accounts. Through this endpoint, only 3rd party-based and confirmed email-based user accounts are listed. It's also important to keep in mind that, by default, only users belonging to the same apps as the requesting user will be shown - you can use [filters](#filters337) to modify this behavior.
```
Comment (ethanroday) [inline source/includes/api/_user_management.md]: Only 3rd party and email based...as opposed to what?

**Review by ethanroday — CHANGES_REQUESTED** [review]
Overall looks good. See comments for specific questions.

File: source/includes/api/_user_management.md (hunk 3 above)
Comment (tiagojsag) [reply source/includes/api/_user_management.md:29]: By "throughout these docs" I assume you mean the docs as a whole, not just this section. Based on what I've seen (and documented) so far, the core elements (dataset, layer, widget, metadata) see to be consistent with the role description provided here. As for other microservices, like area or subscription, that also have a CRUD base, I am less familiar with, so I can't really say - maybe @ikas knows more than I do, as he's been working on those.

I assume you are trying to evaluate if we can havea single definition of the permissions of each role, that then applies to the API as a whole, correct? At this point, I feel like we don't know enough about several MS to build that reliably. Perhaps that's something we can tackle specifically in a future task, and ideally even make that standard part of the dev guidelines, so future development respects that.

File: source/includes/api/_user_management.md (hunk 4 above)
Comment (tiagojsag) [reply source/includes/api/_user_management.md:158]: No, it's specially crafted for these docs only, and doesn't match any existing user

File: source/includes/api/_user_management.md (hunk 5 above)
Comment (tiagojsag) [reply source/includes/api/_user_management.md:271]: You mean the HTTP error code for both scenarios? yes, 422 would fit both scenarios, IMO 
https://www.keycdn.com/support/422-unprocessable-entity

File: source/includes/api/_user_management.md (hunk 6 above)
Comment (tiagojsag) [reply source/includes/api/_user_management.md]: As opposed to those pending confirmation via the link sent via email on registration. I'll improve the text, thanks!

File: source/includes/api/_user_management.md (hunk 3 above)
Comment (ethanroday) [reply source/includes/api/_user_management.md:29]: Okay thanks. I'll consider including this in the next contract.

**Review by ethanroday — APPROVED** [review]

**Pull request merged**